### PR TITLE
Validate pipeline config

### LINKED ISSUE #1836: Seemingly we are not validating colliding "oneOf" fields under artifact
Based on #1649  - there is an example that does not fail with the following: 

```
apiVersion: skaffold/v1beta4
kind: Config
build:
  artifacts:
  - image: acmeprivate.azurecr.io/blah/whatever/application
    jibMaven: {}
    context: api-app # Multi-module Maven project, api-app is the app module
    docker:
      dockerfile: Dockerfile
  local: {}
```

where jibMaven and docker should never be able to coexist on the same artifact.

### PR DESCRIPTION
**Pull request by corneliusweig**
Fields annotated with 'oneOf' should only be specified once by the user. If conflicting fields are set for a given struct, Skaffold now fails with an error.

Close #1836

### CODE REVIEW

**Comment by balopat** [comment]
Awesome, thank you! Will have a deeper look tomorrow.

**Comment by balopat** [comment]
bunch of errors on integration tests:

````
=== RUN   TestBuild/git_tagger
time="2019-03-27T02:04:04Z" level=info msg="[skaffold build -p gitCommit]"
Complete in 1.188978ms
panic: reflect: Field index out of bounds

goroutine 1 [running]:
reflect.(*structType).Field(0x14d6600, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/type.go:1196 +0x1f4
reflect.(*rtype).Field(0x14d6600, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/type.go:931 +0x8c
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x14d6600, 0x2640598, 0x199, 0x14d6600, 0x2640598)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:53 +0x1f9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x13e0800, 0x2640598, 0x96, 0x13e0800, 0x2640598)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:94 +0xdb8
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15b3ca0, 0xc00000e848, 0x99, 0x15b3ca0, 0xc00000e848)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15b3a60, 0xc00000e830, 0x99, 0x15b3a60, 0xc00000e830)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15f3e40, 0xc00000e820, 0x199, 0x15f3e40, 0xc00000e820)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x1413f20, 0xc0003042e8, 0x97, 0x1413f20, 0xc0003042e8)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:83 +0xb76
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15f3f40, 0xc000304240, 0x199, 0x15f3f40, 0xc000304240)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x14e1140, 0xc000304000, 0xc0002b0250, 0x10, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:94 +0xdb8
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.ValidateSchema(0xc000304000, 0xd, 0x4f5601)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:31 +0x4c
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.newRunner(0x26219c0, 0x43498e, 0xc000517b28, 0x455d80, 0xc000517b70)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/runner.go:47 +0x92
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.createRunnerAndBuild(0x18a0d80, 0xc000039a00, 0x1883620, 0xc00000c018, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:103 +0x5e
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.runBuild(0x1883620, 0xc00000c018, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:86 +0x15e
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.NewCmdBuild.func1(0xc0003c3180, 0xc00039c760, 0x0, 0x2, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:53 +0x60
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).execute(0xc0003c3180, 0xc00039c740, 0x2, 0x2, 0xc0003c3180, 0xc00039c740)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0003c2280, 0xc0003c2c80, 0xc0003c2a00, 0xc0003c2780)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).Execute(0xc0003c2280, 0xc00000c018, 0x1883620)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x1883620, 0xc00000c018, 0x1883620, 0xc00000c020, 0xc00009a058, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/skaffold.go:35 +0xa5
main.main()
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/skaffold.go:30 +0x4e
=== RUN   TestBuild/sha256_tagger
time="2019-03-27T02:04:04Z" level=info msg="[skaffold build -p sha256]"
Complete in 1.197565ms
panic: reflect: Field index out of bounds

goroutine 1 [running]:
reflect.(*structType).Field(0x14d6600, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/type.go:1196 +0x1f4
reflect.(*rtype).Field(0x14d6600, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/type.go:931 +0x8c
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x14d6600, 0x2640598, 0x199, 0x14d6600, 0x2640598)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:53 +0x1f9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x13e0800, 0x2640598, 0x96, 0x13e0800, 0x2640598)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:94 +0xdb8
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15b3ca0, 0xc00013a9e8, 0x99, 0x15b3ca0, 0xc00013a9e8)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15b3a60, 0xc00013a9d0, 0x99, 0x15b3a60, 0xc00013a9d0)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15f3e40, 0xc00013a9c0, 0x199, 0x15f3e40, 0xc00013a9c0)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x1413f20, 0xc0002fe2e8, 0x97, 0x1413f20, 0xc0002fe2e8)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:83 +0xb76
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15f3f40, 0xc0002fe240, 0x199, 0x15f3f40, 0xc0002fe240)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x14e1140, 0xc0002fe000, 0xc00014cca0, 0x10, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:94 +0xdb8
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.ValidateSchema(0xc0002fe000, 0xd, 0x4f5601)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:31 +0x4c
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.newRunner(0x26219c0, 0x43498e, 0xc0004f1b28, 0x455d80, 0xc0004f1b70)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/runner.go:47 +0x92
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.createRunnerAndBuild(0x18a0d80, 0xc00016a800, 0x1883620, 0xc00000c018, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:103 +0x5e
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.runBuild(0x1883620, 0xc00000c018, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:86 +0x15e
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.NewCmdBuild.func1(0xc000403900, 0xc000148640, 0x0, 0x2, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:53 +0x60
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).execute(0xc000403900, 0xc000148620, 0x2, 0x2, 0xc000403900, 0xc000148620)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000402280, 0xc000403400, 0xc000403180, 0xc000402c80)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).Execute(0xc000402280, 0xc00000c018, 0x1883620)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x1883620, 0xc00000c018, 0x1883620, 0xc00000c020, 0xc00009a058, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/skaffold.go:35 +0xa5
main.main()
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/skaffold.go:30 +0x4e
=== RUN   TestBuild/dateTime_tagger
time="2019-03-27T02:04:04Z" level=info msg="[skaffold build -p dateTime]"
Complete in 1.326956ms
panic: reflect: Field index out of bounds

goroutine 1 [running]:
reflect.(*structType).Field(0x14d6600, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/type.go:1196 +0x1f4
reflect.(*rtype).Field(0x14d6600, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/type.go:931 +0x8c
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x14d6600, 0x2640598, 0x199, 0x14d6600, 0x2640598)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:53 +0x1f9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x13e0800, 0x2640598, 0x96, 0x13e0800, 0x2640598)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:94 +0xdb8
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15b3ca0, 0xc00012b138, 0x99, 0x15b3ca0, 0xc00012b138)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15b3a60, 0xc00012b120, 0x99, 0x15b3a60, 0xc00012b120)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15f3e40, 0xc00012b110, 0x199, 0x15f3e40, 0xc00012b110)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x1413f20, 0xc00020c2e8, 0x97, 0x1413f20, 0xc00020c2e8)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:83 +0xb76
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15f3f40, 0xc00020c240, 0x199, 0x15f3f40, 0xc00020c240)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x14e1140, 0xc00020c000, 0xc00024c250, 0x10, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:94 +0xdb8
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.ValidateSchema(0xc00020c000, 0xd, 0x4f5601)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:31 +0x4c
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.newRunner(0x26219c0, 0x43498e, 0xc0004dfb28, 0x455d80, 0xc0004dfb70)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/runner.go:47 +0x92
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.createRunnerAndBuild(0x18a0d80, 0xc00001c540, 0x1883620, 0xc00000c018, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:103 +0x5e
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.runBuild(0x1883620, 0xc00000c018, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:86 +0x15e
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.NewCmdBuild.func1(0xc0003bd900, 0xc000138560, 0x0, 0x2, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:53 +0x60
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).execute(0xc0003bd900, 0xc000138540, 0x2, 0x2, 0xc0003bd900, 0xc000138540)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0003bc280, 0xc0003bd400, 0xc0003bd180, 0xc0003bcc80)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).Execute(0xc0003bc280, 0xc00000c018, 0x1883620)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x1883620, 0xc00000c018, 0x1883620, 0xc00000c020, 0xc00009a058, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/skaffold.go:35 +0xa5
main.main()
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/skaffold.go:30 +0x4e
=== RUN   TestBuild/envTemplate_tagger
time="2019-03-27T02:04:04Z" level=info msg="[skaffold build -p envTemplate]"
Complete in 852.187µs
panic: reflect: Field index out of bounds

goroutine 1 [running]:
reflect.(*structType).Field(0x14d6600, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/type.go:1196 +0x1f4
reflect.(*rtype).Field(0x14d6600, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/type.go:931 +0x8c
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x14d6600, 0x2640598, 0x199, 0x14d6600, 0x2640598)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:53 +0x1f9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x13e0800, 0x2640598, 0x96, 0x13e0800, 0x2640598)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:94 +0xdb8
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15b3ca0, 0xc00012da28, 0x99, 0x15b3ca0, 0xc00012da28)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15b3a60, 0xc00012da10, 0x99, 0x15b3a60, 0xc00012da10)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15f3e40, 0xc00012da00, 0x199, 0x15f3e40, 0xc00012da00)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x1413f20, 0xc000026468, 0x97, 0x1413f20, 0xc000026468)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:83 +0xb76
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x15f3f40, 0xc0000263c0, 0x199, 0x15f3f40, 0xc0000263c0)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:69 +0x747
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateOneOf(0x14e1140, 0xc000026180, 0xc0004c5210, 0x10, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:94 +0xdb8
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.ValidateSchema(0xc000026180, 0xd, 0x4f5601)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:31 +0x4c
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.newRunner(0x26219c0, 0x43498e, 0xc0002dfb28, 0x455d80, 0xc0002dfb70)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/runner.go:47 +0x92
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.createRunnerAndBuild(0x18a0d80, 0xc000039a80, 0x1883620, 0xc00000c018, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:103 +0x5e
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.runBuild(0x1883620, 0xc00000c018, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:86 +0x15e
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.NewCmdBuild.func1(0xc00035af00, 0xc0002a8540, 0x0, 0x2, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/build.go:53 +0x60
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).execute(0xc00035af00, 0xc0002a8520, 0x2, 0x2, 0xc00035af00, 0xc0002a8520)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00035a000, 0xc00035a780, 0xc00035a500, 0xc00035a280)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).Execute(0xc00035a000, 0xc00000c018, 0x1883620)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x1883620, 0xc00000c018, 0x1883620, 0xc00000c020, 0xc000088058, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/skaffold.go:35 +0xa5
main.main()
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/skaffold.go:30 +0x4e
--- FAIL: TestBuild (6.83s)
    --- PASS: TestBuild/docker_build (6.69s)
    --- FAIL: TestBuild/git_tagger (0.03s)
        build_test.go:59: skaffold build: exit status 2
    --- FAIL: TestBuild/sha256_tagger (0.04s)
        build_test.go:59: skaffold build: exit status 2
    --- FAIL: TestBuild/dateTime_tagger (0.03s)
        build_test.go:59: skaffold build: exit status 2
    --- FAIL: TestBuild/envTemplate_tagger (0.03s)
        build_test.go:59: skaffold build: exit status 2
```

**Comment by corneliusweig** [comment]
@balopat My bad, forgot that some structs may have no fields.

File: pkg/skaffold/schema/validation/validation_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	}{
+		{
+			name: "only one field set",
+			input: &latest.BuildType{
+				Cluster: &latest.ClusterDetails{},
+			},
+			shouldErr: false,
+		},
+		{
+			name: "only one field set",
```
Comment (balopat) [inline pkg/skaffold/schema/validation/validation_test.go]:
```suggestion
			name: "two colliding buildTypes",
```

File: pkg/skaffold/schema/validation/validation_test.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+			input: &latest.BuildType{
+				GoogleCloudBuild: &latest.GoogleCloudBuild{},
+				Cluster:          &latest.ClusterDetails{},
+			},
+			shouldErr: true,
+			expected:  []string{"googleCloudBuild cluster"},
+		},
+		{
+			name: "deployType with two fields ",
```
Comment (balopat) [inline pkg/skaffold/schema/validation/validation_test.go]:
```suggestion
			name: "deployType with two fields",
```

File: pkg/skaffold/schema/validation/validation_test.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+			input: &latest.DeployType{
+				HelmDeploy:    &latest.HelmDeploy{},
+				KubectlDeploy: &latest.KubectlDeploy{},
+			},
+			shouldErr: true,
+			expected:  []string{"helm kubectl"},
+		},
+		{
+			name: "deployType with three fields ",
```
Comment (balopat) [inline pkg/skaffold/schema/validation/validation_test.go]:
```suggestion
			name: "deployType with three fields",
```
also not sure about the value of this one - we already covered this in the previous testcase?

File: pkg/skaffold/schema/validation/validation_test.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+import (
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+var (
+	pipelineConfig = &latest.SkaffoldPipeline{
```
Comment (balopat) [inline pkg/skaffold/schema/validation/validation_test.go]:
```suggestion
	cfgWithErrors = &latest.SkaffoldPipeline{
```

**Comment by corneliusweig** [comment]
@balopat nits are done. I also added one more commit, moving the validation down after profile application. It occured to me that this could also create `oneOf` conflicts.

**Pull request merged**